hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieWrapperFileSystem.java

```diff
@@ -145,8 +145,11 @@ public static Path convertPathWithScheme(Path oldPath, String newScheme) {
     URI oldURI = oldPath.toUri();
     URI newURI;
     try {
-      newURI = new URI(newScheme, oldURI.getUserInfo(), oldURI.getHost(), oldURI.getPort(), oldURI.getPath(),
-          oldURI.getQuery(), oldURI.getFragment());
+      newURI = new URI(newScheme,
+                oldURI.getAuthority(),
+                oldURI.getPath(),
+                oldURI.getQuery(),
+                oldURI.getFragment());
```

Comment:
can you help clarify the impact of this change pls? it fixes s3 compatibilities, but will it stay fully compatible with previous method usage? not very clear about the nuances of these 2 APIs, and how it caused problems with s3. would like to have some detailed info in the PR to analyze the impact.

Reply:
Yes. It will stay fully compatible. The issue is introduced because of the use of **getHost()** API. This works fine for HDFS like URIs that have a host and a port. Also it works well for most bucket naming conventions that match java URI standards. But for certain patters as explained in the Jira the **getHost()** API breaks because of the way it tries to parse the hostname. Instead this constructor works better because it uses **getAuthority** which correctly extracts bucket name in case of S3. I do not see a need for specifically using the **getHotst/getPort** API, when we have **getAuthority** that can essentially do the same.

Reply:
Thanks for clarifying.